core/bug_api.php

```diff
@@ -508,6 +508,16 @@ public function validate( $p_update_extended = true ) {
 			category_ensure_exists( $this->category_id );
 		}
 
+		if ( $this->status == RESOLVED ) {
+			if ( is_blank( $this->fixed_in_version ) && !is_blank( $this->target_version ) && config_get( 'bug_fixed_in_version_from_target_version', null, null, $this->project_id ) != OFF ) {
+				$this->fixed_in_version = $this->target_version;
+			}
+			if ( is_blank( $this->fixed_in_version ) && config_get( 'bug_fixed_in_version_required', null, null, $this->project_id ) != OFF ) {
+				error_parameters( lang_get( 'fixed_in_version' ) );
+				trigger_error( ERROR_EMPTY_FIELD, ERROR );
+			}
+		}
```

Comment:
There is some duplicated logic here vs changes in bug_change_status_page.php, see if it would be possible avoid that by moving that to an API function somehow.

Reply:
> the main check is server-side 

Yes I understand that, but you are not addressing my point about code duplication. 

It's not blocking, but if you are not going to do something about it, then at least a comment should be added here as well as in common.js, to warn future developers that any changes to the logic must be applied to the other location.